src/manifolds/Flag.jl

```diff
@@ -0,0 +1,190 @@
+    > vol. 194, no. 1, pp. 621–660, Jul. 2022,
+    > doi: [10.1007/s10107-021-01640-3](https://doi.org/10.1007/s10107-021-01640-3).
+"""
+struct Flag{N,dp1} <: AbstractDecoratorManifold{ℝ}
+    subspace_dimensions::ZeroTuple{NTuple{dp1,Int}}
+end
+
+function Flag(N, ns::Vararg{Int,I}) where {I}
+    if ns[1] <= 0
+        error("First dimension in sequence (given: $(ns[1])) must be strictly positive.")
+    end
+    for i in 1:(length(ns) - 1)
+        if ns[i] >= ns[i + 1]
+            error("Sequence of dimensions must be strictly increasing, received $ns")
+        end
+    end
+    if ns[end] >= N
+        error(
+            "Last dimension in sequence (given: $(ns[end])) must be strictly lower than N (given: $N).",
+        )
+    end
+    return Flag{N,I + 1}(ZeroTuple(tuple(ns..., N)))
+end
+
+function active_traits(f, ::Flag, args...)
+    return merge_traits(IsEmbeddedManifold())
+end
+
+"""
+    get_embedding(M::Flag)
+
+Get the embedding of the [`Flag`](@ref) manifold `M`, i.e. the [`Stiefel`](@ref) manifold.
+"""
+get_embedding(M::Flag{N,dp1}) where {N,dp1} = Stiefel(N, M.subspace_dimensions[dp1 - 1])
+
+@doc raw"""
+    injectivity_radius(M::Flag)
+    injectivity_radius(M::Flag, p)
+
+Return the injectivity radius on the [`Flag`](@ref) `M`, which is $\frac{π}{2}$.
+"""
+injectivity_radius(::Flag) = π / 2
+injectivity_radius(::Flag, p) = π / 2
+injectivity_radius(::Flag, ::AbstractRetractionMethod) = π / 2
+injectivity_radius(::Flag, p, ::AbstractRetractionMethod) = π / 2
+
+function Base.isapprox(M::Flag, p, X, Y; atol=sqrt(max_eps(X, Y)), kwargs...)
+    return isapprox(norm(M, p, X - Y), 0; atol=atol, kwargs...)
+end
+function Base.isapprox(M::Flag, p, q; atol=sqrt(max_eps(p, q)), kwargs...)
+    X = inverse_retract(M, p, q, PolarInverseRetraction())
+    return isapprox(norm(M, p, X), 0; atol=atol, kwargs...)
+end
+
+function manifold_dimension(M::Flag{N,dp1}) where {N,dp1}
+    dim = 0
+    for i in 1:(dp1 - 1)
+        dim +=
+            (M.subspace_dimensions[i] - M.subspace_dimensions[i - 1]) *
+            (N - M.subspace_dimensions[i])
+    end
+    return dim
+end
+
+function Base.show(io::IO, M::Flag{N}) where {N}
+    print(io, "Flag($(N)")
+    for d_i in M.subspace_dimensions.x[1:(end - 1)]
+        print(io, ", $d_i")
+    end
+    return print(io, ")")
+end
+
+"""
+    orthogonal_tv_to_stiefel(M::Flag, p::OrthogonalPoint, X::OrthogonalTVector)
+
+Convert tangent vector from [`Flag`](@ref) manifold `M` from orthogonal representation to
+Stiefel representation.
+"""
+function orthogonal_tv_to_stiefel(M::Flag, p::OrthogonalPoint, X::OrthogonalTVector)
+    (N, k) = representation_size(M)
+    return p.value * X.value[:, 1:k]
+end
+
+"""
+    stiefel_tv_to_orthogonal(M::Flag, p::AbstractMatrix, X::AbstractMatrix)
+
+Convert tangent vector from [`Flag`](@ref) manifold `M` from Stiefel representation to
+orthogonal representation.
+"""
+function stiefel_tv_to_orthogonal(M::Flag, p::AbstractMatrix, X::AbstractMatrix)
```

Comment:
same as above, nicer with conversions and speficying one of them to be the array-default-representation cf. Hyperbolic with its three representations and Hyperboloid being the default.